extensions-core/kinesis-indexing-service/src/test/java/org/apache/druid/indexing/kinesis/supervisor/KinesisSupervisorTest.java

```diff
@@ -358,6 +358,7 @@ public void testNoInitialStateWithAutoScaleOut() throws Exception
     Thread.sleep(1 * 1000);
     int taskCountAfterScale = supervisor.getIoConfig().getTaskCount();
     Assert.assertEquals(2, taskCountAfterScale);
+    autoscaler.stop();
```

Comment:
This was causing issues randomly because the autoscaler was kept running and was emitting events which was same listener for the whole suite of tests in this class.